File: rust/zkgroup/src/api/profiles/pni_credential_response.rs
//
// Copyright 2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

use crate::common::simple_types::*;
use crate::crypto;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct PniCredentialResponse {
    pub(crate) reserved: ReservedBytes,
    pub(crate) blinded_credential: crypto::credentials::BlindedPniCredential,
    pub(crate) proof: crypto::proofs::PniCredentialIssuanceProof,
}

File: java/shared/java/org/signal/libsignal/zkgroup/profiles/PniCredentialResponse.java
//
// Copyright 2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.zkgroup.profiles;

import org.signal.libsignal.zkgroup.InvalidInputException;
import org.signal.libsignal.zkgroup.internal.ByteArray;
import org.signal.libsignal.internal.Native;

public final class PniCredentialResponse extends ByteArray {
  public PniCredentialResponse(byte[] contents) throws InvalidInputException {
    super(contents);
    Native.PniCredentialResponse_CheckValidContents(contents);
  }
}
